### DISCUSSION TITLE
Les paramètres d'URL sont supprimés lors du passage de return_url au serveur dans NextJS

### DISCUSSION THREAD

**Discussion by Sokratesli** [PayGreen/documentation#38] [Q&A]
Je rencontre un problème en passant une return_url contenant des paramètres de requête à mon serveur dans une application Next.js 14. Il semble que la return_url soit nettoyée ou tronquée (après un paiement réussi), ce qui entraîne la perte d'une partie de l'URL après son passage.

**Voici un exemple simplifié :**
Je génère l'URL suivante pour la redirection :

```
const return_url = `${data.returnUrl}/.../success?orderNr=${orderNumber}&po_id=${poId}`;
console.log(return_url); 
```
Dans la console, cela affiche correctement :
`http://.../success?orderNr=XYZ123&po_id=po_xyz1234`

Cependant, lorsque je passe cette return_url au serveur pour traitement, il semble que les paramètres après success soient supprimés. Plus précisément, les paramètres orderNr et po_id ne sont pas renvoyés comme prévu.

**Comportement attendu :**
Je m'attends à ce que l'URL entière, y compris les paramètres de requête, soit préservée lors de son passage au serveur.

**Comportement actuel :**
Seule la partie jusqu'à /success semble arriver au serveur, et les paramètres (?orderNr=XYZ123&po_id=po_xyz1234) sont manquants.

**Environnement :**
Version de Next.js : 14.x.x

**Logs du serveur :** Aucun message d'erreur spécifique dans les logs, mais les paramètres semblent être supprimés.
Quelqu'un a-t-il déjà rencontré ce problème, et comment puis-je m'assurer que les paramètres de la requête sont conservés lors de l'envoi de la return_url ?

**Comment by gaelDolle** [comment]
Bonjour,

Désolé pour la réponse tardive.

Le po_id et le status du paiement sont ajoutés automatiquement au return_url. Votre orderNr sera lui aussi ajouté.

Dans le cas où vous rencontrez toujours le problème, merci de faire un ticket support afin d'examiner plus spécifiquement votre cas d'usage.

Cordialement
